### PR TITLE
Adding 2018 april schema and editing 2018 Jan schema for IotHub

### PR DESCRIPTION
**Pull request by akumardu**

### CODE REVIEW

File: schemas/2018-01-22/Microsoft.Devices.json [hunk 1]
```diff
@@ -71,6 +71,32 @@
...
+        "name": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.Devices/IotHubs/eventHubEndpoints/ConsumerGroups"
+          ]
+        },
+        "apiVersion": {
```
Comment (TianoMS) [inline schemas/2018-01-22/Microsoft.Devices.json]: >apiVersion [](start = 9, length = 10)

you don't have a "properties" bag?

File: schemas/2018-04-01/Microsoft.Devices.json [hunk 2]
```diff
@@ -0,0 +1,1044 @@
...
+            "Microsoft.Devices/IotHubs/certificates"
+          ]
+        },
+        "apiVersion": {
+          "type": "string",
+          "enum": [
+            "2018-04-01"
+          ]
+        },
+        "certificate": {
```
Comment (TianoMS) [inline schemas/2018-04-01/Microsoft.Devices.json]: >certificate [](start = 9, length = 11)

this is not allowed. al the RP specific properties should go into "properties" bag.

File: schemas/2018-04-01/Microsoft.Devices.json [hunk 3]
```diff
@@ -0,0 +1,1044 @@
...
+        "name": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.Devices/IotHubs/eventHubEndpoints/ConsumerGroups"
+          ]
+        },
+        "apiVersion": {
```
Comment (TianoMS) [inline schemas/2018-04-01/Microsoft.Devices.json]: >apiVersion [](start = 9, length = 10)

no "properties" bag for this resource type.

File: schemas/2018-04-01/Microsoft.Devices.json [hunk 4]
```diff
@@ -0,0 +1,1044 @@
+{
+  "id": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Devices.json#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Microsoft.Devices",
+  "description": "Microsoft Devices Resource Types",
+  "resourceDefinitions": {
+    "IotHubs": {
```
Comment (TianoMS) [inline schemas/2018-04-01/Microsoft.Devices.json]: >IotHubs [](start = 5, length = 7)

tests are missing for the new resource types.

File: schemas/2018-04-01/Microsoft.Devices.json (hunk 2 above)
Comment (akumardu) [reply schemas/2018-04-01/Microsoft.Devices.json]: We did get an exception from Gaurav on Certificates. Should I remove the Certificates from schema?

File: schemas/2018-01-22/Microsoft.Devices.json (hunk 1 above)
Comment (akumardu) [reply schemas/2018-01-22/Microsoft.Devices.json]: Swagger has this as Consumer group info which contains the property bags.
"EventHubConsumerGroupInfo": {
      "description": "The properties of the EventHubConsumerGroupInfo object.",
      "x-ms-azure-resource": true,
      "type": "object",
      "properties": {
        "properties": {
          "description": "The tags.",
          "type": "object",
          "additionalProperties": {
            "type": "string"
          }
        },
        "id": {
          "description": "The Event Hub-compatible consumer group identifier.",
          "type": "string",
          "readOnly": true
        },
        "name": {
          "description": "The Event Hub-compatible consumer group name.",
          "type": "string",
          "readOnly": true
        },
        "type": {
          "description": "the resource type.",
          "type": "string",
          "readOnly": true
        },
        "etag": {
          "description": "The etag.",
          "type": "string",
          "readOnly": true
        }
      }
    }
Does this object have some issue as well?

File: schemas/2018-01-22/Microsoft.Devices.json [hunk 5]
```diff
@@ -71,6 +71,32 @@
...
+          "type": "string",
+          "enum": [
+            "2018-04-01"
```
Comment (TianoMS) [inline schemas/2018-01-22/Microsoft.Devices.json]: >"2018-04-01" [](start = 12, length = 12)

shouldn't this be "2018-01-22"?

File: schemas/2018-04-01/Microsoft.Devices.json [hunk 6]
```diff
@@ -0,0 +1,1005 @@
...
+            "certificates"
+          ]
+        },
+        "apiVersion": {
+          "type": "string",
+          "enum": [
+            "2018-04-01"
+          ]
+        },
+        "certificate": {
```
Comment (TianoMS) [inline schemas/2018-04-01/Microsoft.Devices.json]: >certificate [](start = 9, length = 11)

property is not allowed here.

File: schemas/2018-04-01/Microsoft.Devices.json [hunk 7]
```diff
@@ -0,0 +1,1005 @@
...
+        "isEnabled"
+      ],
+      "description": "The properties of the fallback route. IoT Hub uses these properties when it routes messages to the fallback endpoint."
+    },
+    "FeedbackProperties": {
+      "type": "object",
+      "properties": {
+        "lockDurationAsIso8601": {
+          "type": "string",
+          "format": "duration",
```
Comment (TianoMS) [inline schemas/2018-04-01/Microsoft.Devices.json]: >duration [](start = 21, length = 8)

duration format is not supported by Visual Studio. I guess you can remove them.

File: tests/2018-04-01/Microsoft.Devices.tests.json [hunk 8]
```diff
@@ -0,0 +1,2106 @@
...
+        "sku": {
+          "name": "F1",
+          "capacity": 1
+        },
+        "properties": {
+          "authorizationPolicies": [
+            {
+              "keyName": "IotHubAllAccessSasKey",
+              "primaryKey": "CQN2K33r45/0WeIjpqmErV5EIvX8JZrozt3NEHCEkG8=",
+              "secondaryKey": "RpkT42rNZl4/J08pI8t
```
Comment (TianoMS) [inline tests/2018-04-01/Microsoft.Devices.tests.json]: >IotHubs - RoutingProperty [](start = 15, length = 25)

Some test cases are failing. Please fix them.

File: schemas/2018-04-01/Microsoft.Devices.json (hunk 7 above)
Comment (akumardu) [reply schemas/2018-04-01/Microsoft.Devices.json]: Ack.

**Pull request merged**